internal/util.ml

```diff
@@ -2128,34 +2137,56 @@ let child_of_parent conf base p =
 
 
 (* ************************************************************************** *)
-(*  [Fonc] husband_wife : config -> base -> person -> unit                    *)
+(*  [Fonc] husband_wife : config -> base -> person -> bool -> string                    *)
 (** [Description] : Traduction selon l'existence du premier conjoint
                     différent de ?? :
                       * époux/épouse de Jean/Jeanne
     [Args] :
       - conf : configuration
       - base : base de donnée
       - p    : person
+      - all  : if true, list all spouses
     [Retour] : string
-    [Rem] : Non exporté en clair hors de ce module.                           *)
+    [Rem]    : Exporté en clair hors de ce module.                           *)
 (* ************************************************************************** *)
-let husband_wife conf base p =
-  let rec loop i =
-    if i < Array.length (get_family p) then
-      let fam = foi base (get_family p).(i) in
-      let conjoint = Gutil.spouse (get_key_index p) fam in
-      let conjoint = pget conf base conjoint in
-      if p_first_name base conjoint <> "?" || p_surname base conjoint <> "?"
-      then
-        let relation =
-          Printf.sprintf (relation_txt conf (get_sex p) fam) (fun () -> "")
-        in
-        translate_eval (relation ^ " " ^ person_text conf base conjoint)
-      else loop (i + 1)
-    else ""
+let husband_wife conf base p all =
+  let relation =
+    let rec loop i =
+      if i < Array.length (get_family p) then
+        let fam = foi base (get_family p).(i) in
+        let conjoint = Gutil.spouse (get_key_index p) fam in
+        let conjoint = pget conf base conjoint in
+        if p_first_name base conjoint <> "?" || p_surname base conjoint <> "?"
```

Comment:
Could we turn (in util.ml)
```ocaml
let init_cache_info conf base =
  let nb_ind = Gwdb.nb_of_persons base in
  (* Reset le nombre réel de personnes d'une base. *)
  let nb_real_persons = ref 0 in
  let is_empty_name p =
    (is_empty_string (get_surname p) || is_quest_string (get_surname p)) &&
    (is_empty_string (get_first_name p) || is_quest_string (get_first_name p))
  in
  for i = 0 to nb_ind - 1 do
    let ip = Adef.iper_of_int i in
    let p = poi base ip in
    if is_empty_name p then () else incr nb_real_persons
  done;
  let () =
    Hashtbl.add ht_cache_info cache_nb_base_persons
      (string_of_int !nb_real_persons)
  in
  write_cache_info conf
```
into
```ocaml
let is_empty_name p =
  (is_empty_string (get_surname p) || is_quest_string (get_surname p)) &&
  (is_empty_string (get_first_name p) || is_quest_string (get_first_name p))

let init_cache_info conf base =
  let nb_ind = Gwdb.nb_of_persons base in
  (* Reset le nombre réel de personnes d'une base. *)
  let nb_real_persons = ref 0 in
  for i = 0 to nb_ind - 1 do
    let ip = Adef.iper_of_int i in
    let p = poi base ip in
    if is_empty_name p then () else incr nb_real_persons
  done;
  let () =
    Hashtbl.add ht_cache_info cache_nb_base_persons
      (string_of_int !nb_real_persons)
  in
  write_cache_info conf
```

and use `is_empty_name` instead of these `<> "?"` comparisons?

It should achieve the same goal, but it is cleaner and also more efficient.